sidecar_acceptor.go

```diff
@@ -131,3 +193,268 @@ func (a *SidecarAcceptor) RegisterSidecar(ctx context.Context,
+// given sidecar ticket. We subscribe to auction events using the multisig key
+// we gave out when we registered the ticket.
+func (a *SidecarAcceptor) ExpectChannel(ctx context.Context,
+	t *sidecar.Ticket) error {
+
+	a.Lock()
+	defer a.Unlock()
+
+	if t.Order == nil {
+		return fmt.Errorf("order in sidecar ticket is missing")
+	}
+
+	nonce := t.Order.BidNonce
+	_, ok := a.pendingSidecarOrders[nonce]
+	if ok {
+		return fmt.Errorf("sidecar with order nonce %x is already "+
+			"registered", nonce[:])
+	}
+
+	// We didn't know about this ticket for this nonce before so let's now
+	// update its state in the database and start expecting a channel for it
+	// now.
+	t.State = sidecar.StateExpectingChannel
+	if err := a.store.UpdateSidecar(t); err != nil {
+		return fmt.Errorf("error updating sidecar: %v", err)
+	}
+
+	a.pendingSidecarOrders[nonce] = t
+
+	return a.client.SubscribeAccountUpdates(ctx, &keychain.KeyDescriptor{
+		KeyLocator: keychain.KeyLocator{
+			Family: keychain.KeyFamilyMultiSig,
+			Index:  t.Recipient.MultiSigKeyIndex,
+		},
+		PubKey: t.Recipient.MultiSigPubKey,
+	})
+}
+
+// HasPendingBatch returns true if the acceptor is currently working on
+// accepting a channel from a pending batch.
+func (a *SidecarAcceptor) HasPendingBatch() bool {
+	a.Lock()
+	defer a.Unlock()
+
+	return a.pendingBatch != nil
+}
+
+// matchPrepare handles an incoming OrderMatchPrepare message from the server.
+// Since we're only on the receiving end of a sidecar channel (which is always
+// a bid order) the tasks are simplified compared to normal bid order execution.
+func (a *SidecarAcceptor) matchPrepare(
+	msg *auctioneerrpc.OrderMatchPrepare) error {
+
+	// Parse and formally validate what we got from the server.
+	log.Tracef("Received prepare msg from server, batch_id=%x: %v",
+		msg.BatchId, spew.Sdump(msg))
+	batch, err := order.ParseRPCBatch(msg)
+	if err != nil {
+		// If we aren't able to parse the batch for some reason, then
+		// we'll send a reject message.
+		log.Error("unable to parse batch: %v", err)
+		return a.sendRejectBatch(batch, err)
+	}
+
+	log.Infof("Received PrepareMsg for batch=%x, num_orders=%v",
+		batch.ID[:], len(batch.MatchedOrders))
+
+	a.Lock()
+	pendingBatch := a.pendingBatch
+	a.Unlock()
+
+	// If there is still a pending batch around from a previous iteration,
+	// we need to clean up the pending channels first.
+	if pendingBatch != nil {
+		if err := a.removeShims(pendingBatch); err != nil {
+			log.Error("unable to cleanup previous batch: %v", err)
+			return a.sendRejectBatch(batch, err)
+		}
+	}
+
+	// Before we accept the batch, we'll finish preparations on our end
+	// which include applying any order match predicates, connecting out to
+	// peers, and registering funding shim. We don't do a full batch
+	// validation since we don't have any information about the account
+	// that's being used to pay for the sidecar channel.
+	err = a.fundingManager.PrepChannelFunding(
```

Comment:
Ahh I see it's called here indirectly, and we only needed to derive the funding shims back then in order to know what to expect.